### PR TITLE
Show custom validity message only when field is empty.

### PR DESCRIPTION
**Pull request by ADmad**
This allows browser to still show its default message
for other validation failures like invalid string for email field.

### CODE REVIEW

**Comment by ADmad** [comment]
https://jsfiddle.net/oembq2rw/

**Comment by ADmad** [comment]
Also change the context classes to return "notEmpty" message instead of "requirePresence" message as I explained here https://github.com/cakephp/cakephp/pull/13498#issuecomment-519908619.

File: src/View/Helper/FormHelper.php [hunk 1]
```diff
@@ -1345,7 +1345,7 @@ protected function _magicOptions(string $fieldName, array $options, bool $allowO
 
             if ($this->getConfig('autoSetCustomValidity')) {
                 $options['oninvalid'] = "this.setCustomValidity(''); "
-                    . "if (!this.validity.valid) this.setCustomValidity('$message')";
+                    . "if (!this.value) this.setCustomValidity('$message')";
```
Comment (markstory) [inline src/View/Helper/FormHelper.php]: Will this still take HTML5 pattern validations into account?

File: tests/TestCase/View/Form/FormContextTest.php [hunk 2]
```diff
@@ -53,7 +53,7 @@ public function setUp(): void
     public function testGetRequiredMessage()
     {
         $validator = new Validator();
-        $validator->requirePresence('title', true, 'Don\'t forget a title!');
+        $validator->notEmpty('title', 'Don\'t forget a title!');
```
Comment (markstory) [inline tests/TestCase/View/Form/FormContextTest.php]: Seems like this entire test could be removed if it is for testing requirePresence() interactions with message generation.

File: src/View/Helper/FormHelper.php (hunk 1 above)
Comment (ADmad) [reply src/View/Helper/FormHelper.php]: I updated the fiddle. As for the email field the custom validity message is shown only when field is empty. When failing to match the pattern it shows bower's default message.

File: tests/TestCase/View/Form/FormContextTest.php (hunk 2 above)
Comment (ADmad) [reply tests/TestCase/View/Form/FormContextTest.php]: The test is required. Istead of requirePresence() message it does need to check for notEmpty() message. There are no other tests for the same.

**Pull request merged**